pkg/rdkafka/Symfony/RdKafkaTransportFactory.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Enqueue\RdKafka\Symfony;
+
+use Enqueue\RdKafka\Client\RdKafkaDriver;
+use Enqueue\RdKafka\RdKafkaConnectionFactory;
+use Enqueue\RdKafka\RdKafkaContext;
+use Enqueue\Symfony\DriverFactoryInterface;
+use Enqueue\Symfony\TransportFactoryInterface;
+use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+class RdKafkaTransportFactory implements TransportFactoryInterface, DriverFactoryInterface
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @param string $name
+     */
+    public function __construct($name = 'rdkafka')
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addConfiguration(ArrayNodeDefinition $builder)
+    {
+        $builder
+            ->beforeNormalization()
+                ->ifString()
+                ->then(function ($v) {
+                    return ['dsn' => $v];
+                })
+            ->end()
+            ->fixXmlConfig('topic')
+            ->children()
+                ->scalarNode('dsn')
+                    ->info('The kafka DSN. Other parameters are ignored if set')
+                ->end()
+                ->arrayNode('global')
+                    ->children()
+                        ->scalarNode('metadata.broker.list')->end()
```

Comment:
Symfony does not allow dots in configuration nodes but the [librdkafka configuration keys](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md) do contain them, need to figure out how to maybe normalize them?

Reply:
you can define a variable node (which allows everything to be put in it).

Reply:
As far as i can tell even a variable node requires a name and will [throw an exception](https://github.com/symfony/config/blob/v4.0.8/Definition/BaseNode.php#L42) when it contains any dots.

I can just remove the `metadata.broker.list` scalarNode and just leave the `global` ArrayNode or . Any suggestions?

Reply:
By default you can configure the broker over the config in this way:
```
enqueue:
    transport:
        default: 'kafka://localhost:9092'
```
So I think it is a good solution to provide a alternative config way like this:
```
enqueue:
    transport:
        kafka:
            metadata_broker_list: localhost:9092
            ...
```

The easier way is to use `->scalarNode('metadata_broker_list')->end()`

Reply:
@dergriewatz As the configuration is passed to rdkafka/librdkafka when creating a new Producer or Consumer and the global configuration properties are pretty clearly defined as having dots (check my first link) so i don't think it will work.

Reply:
I went with the variable node for now as parsing or normalizing the configuration nodes would require maintaining a map of all rdkafka configuration properties (as it consists of keys with both dots and underscores). And i am not particularly in favour of this.